### DISCUSSION TITLE
GCHandle.Alloc(x, GCHandleType.Pinned) calls should be replaced by fixed statements with IntPtr casts where appropriate

### DISCUSSION THREAD

**Discussion by Mrnikbobjeff** [dotnet/maui#22620] [Ideas]
GCHandle.Alloc for pinned statements is strictly worse than using a fixed statement with a T* cast. The roslyn lowering does not incur the same penalty as the GCHandle. I was told to ping @jonathanpeppers for the performance topic

**Comment by Mrnikbobjeff** [comment]
After further sighting of the code I believe this would also reduce risks due to automatic unpinning in exception cases due to the finally lowering

**Reply by jonathanpeppers** [reply]
Can you link to the code you are suggesting to change here? We have no idea what might be using `GCHandle` without searching for it.

I know `GCHandle` is used internally by Objective-C bindings.
